layers/+spacemacs/spacemacs-editing/packages.el

```diff
@@ -321,6 +321,25 @@ It will toggle the overlay under point or create an overlay of one character."
       (sp-pair "{" nil :post-handlers
                '(:add (spacemacs/smartparens-pair-newline-and-indent "RET")))
       (sp-pair "[" nil :post-handlers
-               '(:add (spacemacs/smartparens-pair-newline-and-indent "RET"))))))
-
+               '(:add (spacemacs/smartparens-pair-newline-and-indent "RET")))
 
+      (defun spacemacs/smart-closing-parenthesis ()
+        (interactive)
+        (let* ((sp-navigate-close-if-unbalanced t)
+               (current-pos (point))
+               (current-line (what-line))
```

Comment:
`(what-line)` is meant for interactive use, you probably want `(line-number-at-pos)` here.